knode/validator/validator.go

```diff
@@ -642,9 +649,15 @@ func (val *validator) updateValidators(checksum [32]byte, genesis bool) error {
 }
 
 func (val *validator) Deposits() ([]*types.Deposit, error) {
+	if !val.Validating() {
```

Comment:
same here

Reply:
If a validator is not validating the next line `return val.consensus.Deposits(val.walletAccount.Account().Address)` will panic. We can check deposits only if in validating state.

Reply:
I tough that deposits is just a contract that you ask for a list?

Reply:
It's true but it uses `val.walletAccount.Account()` that is not `nil` only after `Start()`. How do you think, maybe we can make a new issue to rewrite this piece of code to use Deposits without Start pre-requirements?

Reply:
yeah, I guess, this is God like class

Reply:
Done https://github.com/kowala-tech/kcoin/issues/392

Reply:
Any user should be able to check their deposits even if it's not validating.

Reply:
I think so. I've rewritten `Deposits()` in the last commit.